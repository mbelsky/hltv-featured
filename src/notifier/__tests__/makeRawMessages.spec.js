const { makeRawMessages } = require('../makeRawMessages')

const data = {
  matches: [
    {
      stars: 0,
      id: 2341120,
      title: '0 Renegades vs NaVi',
      href: 'https://www.hltv.org/matches/2341120',
      unixTimestamp: 1589090400000,
      event: 'ESL One: Road to Rio - Oceania',
    },
    {
      stars: 5,
      id: 2341125,
      title: '5 Renegades vs ORDER',
      href: 'https://www.hltv.org/matches/2341125',
      unixTimestamp: 1589090400000,
      event: 'ESL One: Road to Rio - Oceania',
    },
  ],
  timeZoneOffsetsMap: {
    'Asia/Bishkek': 21600,
  },
  users: {
    0: {
      seenCustomizeLocationMessage: true,
      seenEmptyMessage: true,
      filter: 5,
    },
    1: {
      filter: 5,
      location: {
        timeZoneId: 'Asia/Bishkek',
      },
    },
    2: {
      filter: 5,
      favorites: ['navi'],
    },
  },
}

describe('makeRawMessages', () => {
  test('common', () => {
    expect(makeRawMessages(data)).toMatchSnapshot()
  })

  test('without matches', () => {
    expect(
      makeRawMessages({
        ...data,
        matches: [],
      }),
    ).toMatchSnapshot()
  })

  test('without favorite matches', () => {
    const { favoriteTeamsMatches } = makeRawMessages({
      ...data,
      users: {
        0: {
          filter: 0,
          favorites: ['unknown team'],
        },
      },
    })

    expect(favoriteTeamsMatches).toBe(undefined)
  })
})
