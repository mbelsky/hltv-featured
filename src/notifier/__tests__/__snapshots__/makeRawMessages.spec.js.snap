// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeRawMessages common 1`] = `
Array [
  Object {
    "customLocationMessage": undefined,
    "favoriteTeamsMatches": undefined,
    "featuredMatches": Object {
      "message": "<a href=\\"https://www.hltv.org/matches/2341125\\">5Â RenegadesÂ vsÂ ORDER</a>
Rating: â˜†â˜†â˜†â˜†â˜†
<i>10Â MayÂ 06:00Â UTC @ ESL One: Road to Rio - Oceania</i>",
      "type": "SUCCESS",
    },
    "id": "0",
  },
  Object {
    "customLocationMessage": undefined,
    "favoriteTeamsMatches": undefined,
    "featuredMatches": Object {
      "message": "<a href=\\"https://www.hltv.org/matches/2341125\\">5Â RenegadesÂ vsÂ ORDER</a>
Rating: â˜†â˜†â˜†â˜†â˜†
<i>10Â MayÂ 12:00 @ ESL One: Road to Rio - Oceania</i>",
      "type": "SUCCESS",
    },
    "id": "1",
  },
  Object {
    "customLocationMessage": "You may change matches timezone for your feed. Just tap on the ðŸ“Ž attachment button, choose Location and send your city coordinates as a message.",
    "favoriteTeamsMatches": "<b>Your favorite teams matches:</b>

<a href=\\"https://www.hltv.org/matches/2341120\\">0Â RenegadesÂ vsÂ NaVi</a>
Rating: â€“
<i>10Â MayÂ 06:00Â UTC @ ESL One: Road to Rio - Oceania</i>",
    "featuredMatches": Object {
      "message": "<a href=\\"https://www.hltv.org/matches/2341125\\">5Â RenegadesÂ vsÂ ORDER</a>
Rating: â˜†â˜†â˜†â˜†â˜†
<i>10Â MayÂ 06:00Â UTC @ ESL One: Road to Rio - Oceania</i>",
      "type": "SUCCESS",
    },
    "id": "2",
  },
]
`;

exports[`makeRawMessages without matches 1`] = `
Array [
  Object {
    "customLocationMessage": undefined,
    "favoriteTeamsMatches": undefined,
    "featuredMatches": undefined,
    "id": "0",
  },
  Object {
    "customLocationMessage": undefined,
    "favoriteTeamsMatches": undefined,
    "featuredMatches": Object {
      "message": "There are no 5-stars matches today. And I won't spam you with this message every morning.

Next time just send /upcoming if no morning notification.",
      "type": "EMPTY",
    },
    "id": "1",
  },
  Object {
    "customLocationMessage": "You may change matches timezone for your feed. Just tap on the ðŸ“Ž attachment button, choose Location and send your city coordinates as a message.",
    "favoriteTeamsMatches": undefined,
    "featuredMatches": Object {
      "message": "There are no 5-stars matches today. And I won't spam you with this message every morning.

Next time just send /upcoming if no morning notification.",
      "type": "EMPTY",
    },
    "id": "2",
  },
]
`;
